app/code/community/Ebizmarts/MailChimp/Model/Api/Products.php

```diff
@@ -640,12 +596,34 @@ protected function getConfigurableChildrenIds($product)
     }
 
     /**
+     * @param $product
      * @param $magentoStoreId
-     * @return Mage_Catalog_Model_Resource_Product_Collection
+     * @param bool $isBuildUpdateProductRequest
+     * @return array | return an array with the childs of the product passed by parameter
      */
-    protected function makeProductChildrenCollection($magentoStoreId)
+    public function makeProductChildrenCollection($product, $magentoStoreId, $isBuildUpdateProductRequest = false)
```

Comment:
Function is not returning a collection as the title suggest. Please change title to match what is returned.